src/osd/osd_types.cc

```diff
@@ -1310,6 +1310,14 @@ bool pg_pool_t::maybe_updated_removed_snaps(const interval_set<snapid_t>& cached
   return true;
 }
 
+void pg_pool_t::purged_snap(const snapid_t& purged) {
+  deleting_snaps.erase(purged);
```

Comment:
It looks like this needs to check if deleting_snaps contains the purged snap, since I have an OSD that crashes during startup as follows:
```
 ceph version 12.2.0 (32ce2a3ae5239ee33d6150705cdb24d43bab910c) luminous (rc)
 1: (()+0x9fbb3e) [0x562b96be1b3e]
 2: (()+0x10ea0) [0x7f0753f13ea0]
 3: (gsignal()+0x38) [0x7f075328a148]
 4: (abort()+0x16a) [0x7f075328b5ca]
 5: (ceph::__ceph_assert_fail(char const*, char const*, int, char const*)+0x28e) [0x562b96c2489e]
 6: (pg_pool_t::purged_snap(snapid_t const&)+0x245) [0x562b968b64f5]
 7: (PrimaryLogPG::AwaitAsyncWork::react(PrimaryLogPG::DoSnapWork const&)+0x105a) [0x562b96803fca]
 8: (boost::statechart::simple_state<PrimaryLogPG::AwaitAsyncWork, PrimaryLogPG::Trimming, boost::mpl::list<mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::
na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na>, (boost::statechart::history_mode)0>::react_impl(boost::statechart::event_base const&, void const*)+0xca) [0x562b96854a3a]
 9: (boost::statechart::state_machine<PrimaryLogPG::SnapTrimmer, PrimaryLogPG::NotTrimming, std::allocator<void>, boost::statechart::null_exception_translator>::process_event(boost::statechart::event_base const&)+0x56) [0x562b96835146]
 10: (PrimaryLogPG::snap_trimmer(unsigned int)+0x19c) [0x562b967a163c]
 11: (OSD::ShardedOpWQ::_process(unsigned int, ceph::heartbeat_handle_d*)+0x100c) [0x562b9666987c]
 12: (ShardedThreadPool::shardedthreadpool_worker(unsigned int)+0x884) [0x562b96c29504]
 13: (ShardedThreadPool::WorkThreadSharded::entry()+0x10) [0x562b96c2c540]
 14: (()+0x73e4) [0x7f0753f0a3e4]
 15: (clone()+0x6d) [0x7f075333ea9d]
```